protocols/imap/src/test/java/org/apache/james/imap/processor/base/UidMsnMapperTest.java

```diff
@@ -218,6 +218,45 @@ public void removeShouldKeepAValidMappingWhenDeletingMiddle() {
     }
 
     @Test
+    public void addUidShouldCommute() {
```

Comment:
No I would not pretend that, but I show a test case, (of course TDD -> broken before) and solved it.

Maybe you prefer if I say the exact same thing with a non-ensemblist vocabulary: 

**addUidShouldSupportOutOfOrderUpdates**